datastore/elastic_search/query.py

```diff
@@ -336,6 +337,9 @@ def _run_es_search(connection, msearch=False, **kwargs):
         raise ValueError('Scrolling is not supported in msearch mode')
 
     result = connection.search(scroll=scroll, **kwargs)
+    if 'hits' not in result:
+        raise DataStoreRequestException(f'No hits from ES search for kwargs -> {kwargs}')
```

Comment:
Our case usually hits and returns from block 331-334

Instead of this being here, I would suggest wrapping lines 311 and 312 as

```python
response = None
try:
    response = _run_es_search(connection, msearch=True, **kwargs)
    results = _parse_es_search_results(response.get("responses"))
except Exception as e:
    raise DataStoreRequestException(f'Error in datastore query on index: {index}', engine='elasticsearch', query=json.dumps(data), response=json.dumps(response)') from e    
```

then the catching code can explicitly access and log the `e.query` and `e.response` only in the logs.
one reason why I suggest this is because very complicated message strings can be difficult for Sentry to group.

Same comment for this flow in `ner_v2/detectors/textual/queries.py`

---
Note that, even with this approach we will miss out on `response` when `scroll=True` and `hits` is missing in the response because code will break at line 339, but I guess that can be taken up later

Reply:
Yes agree this is a good idea. Also doing this same for `ner_v2/text`